### ISSUE TITLE
Add ability to pass a URL or schema dictionary to TRAPISchemaValidator.validate so it can be used to test example data that corresponds to changes in the ReasonerAPI specification on PR. 

### ISSUE THREAD

**Issue by sierra-moxon** [NCATSTranslator/reasoner-validator#56]
**Is your feature request related to a problem? Please describe.**
When I make changes to the ReasonerAPI schema, I want to write tests that validate the examples I provided with my change. 

Following this basic workflow:
- make changes in the TranslatorReasonerAPI.yaml in a branch
- write some example data that conforms to the new schema
- run some tests to make sure the example data conforms to the new schema.

TRAPISchemaValidator.validate(trapi_version) sets the trapi_version to the latest version of TRAPI (1.3.0) even if I pass in another version.  (e.g. 1.3.1rc1) 

I think this works now because 1.3.0 was made after the last round of schema changes (and ReasonerAPI tests) were in and working.  The tests currently validate because they match the latest released schema, but will fail on each PR that attempts to modify the schema.  

**Describe the solution you'd like**
I'd like to be able to pass a URL (e.g.: https://raw.githubusercontent.com/sierra-moxon/ReasonerAPI/mkg_qualifiers/TranslatorReasonerAPI.yaml), or the result of a yaml.load operation (a dictionary containing the schema I want to validate against) to the TRAPISchemaValidator.validate() method for it to use as the schema of reference.  

**Describe alternatives you've considered**
I can write my own validator that uses a local/updated version of the schema to run on PR for ReasonerAPI PRs. 

**Additional Context**
schema to validate against: https://raw.githubusercontent.com/sierra-moxon/ReasonerAPI/mkg_qualifiers/TranslatorReasonerAPI.yaml
simple tests: https://github.com/sierra-moxon/ReasonerAPI/blob/f42e7786e9c958d17ceb04f51ce91c80ad415130/tests/test_valid.py#L56

**Comment by RichardBruskiewich** [comment]
Hi @sierra-moxon, 

Your observations are correct in that if a stated version is not actually an actual ReasonerAPI Git release, it is ignored (or perhaps, some 'close' version substituted in).

The reasoner-validator code does fully vet its TRAPI schemata versions against the list of known ReasonerAPI project releases (see [**reasoner_validator.versioning**](https://github.com/NCATSTranslator/reasoner-validator/blob/master/reasoner_validator/versioning.py)). 

I think that it was actually fixed some time back to handle _extended_ SemVer specifications. BTW, a 'release candidate' should  have a hyphen, i.e. something like **1.3.1-rc1** (or **1.3.1-rc.1**) should work. The full regex is [here](https://github.com/NCATSTranslator/reasoner-validator/blob/master/reasoner_validator/versioning.py#L26) ... a bit complex (adapted  from an official semver example site).

Thus, it should be feasible  to issue a "release candidate" release that should NOT be tagged as 'latest' but which can still then be fed into the reasoner-validator as a valid release for testing. 

Let me know if you think I should still override the above and provide for a direct sample schema URL instead. If so, I can take a look at this sometime this coming week.

**Comment by sierra-moxon** [comment]
I tried 1.3.1-rc.1 in the API yaml file, and I got this error message back (note the print out of the version from the yaml file I am submitting and the trapi_version that results):

```
test_valid.py::test_metakg_examples FAILED                               [100%]1.3.1-rc.1
{'trapi_version': '1.3.0', 'biolink_version': None, 'messages': {'information': [], 'warnings': [], 'errors': []}}
```

Did you mean that I should create a release candidate tag in the ReasonerAPI repo in order for this to work?  I'd like the ability to run the tests locally, using reasoner-validator, without a tag in the repo (just using a branch).

**Comment by RichardBruskiewich** [comment]
> I tried 1.3.1-rc.1 in the API yaml file, and I got this error message back (note the print out of the version from the yaml file I am submitting and the trapi_version that results):
> 
> 
> 
> ```
> 
> test_valid.py::test_metakg_examples FAILED                               [100%]1.3.1-rc.1
> 
> {'trapi_version': '1.3.0', 'biolink_version': None, 'messages': {'information': [], 'warnings': [], 'errors': []}}
> 
> ```
> 
> 
> 
> Did you mean that I should create a release candidate tag in the ReasonerAPI repo in order for this to work?  I'd like the ability to run the tests locally, using reasoner-validator, without a tag in the repo (just using a branch).
Hi Sierra, sorry for the slow reply.
The short answer is again "yes" you currently need an official GitHub release for the current code to retrieve the target schema.

**Comment by RichardBruskiewich** [comment]
Just to be clear, here's the code that generates the list of TRAPI versions: 

```python
# Undocumented possible local environmental variable
# override of the ReasonerAPI schema access endpoint
GIT_ORG = environ.setdefault('GIT_ORGANIZATION', "NCATSTranslator")
GIT_REPO = environ.setdefault('GIT_REPOSITORY', "ReasonerAPI")

response = requests.get(f"https://api.github.com/repos/{GIT_ORG}/{GIT_REPO}/releases")
releases = response.json()
versions = [
    release["tag_name"][1:]
    for release in releases
    if release["tag_name"].startswith("v")
]
```

**Comment by RichardBruskiewich** [comment]
@sierra-moxon PR https://github.com/NCATSTranslator/reasoner-validator/pull/59 now allows Git branch names as the 'release' identifier for TRAPI schemata. I hope this suffices for your use case.  I suspect, in principle, I could take things one more step toward a schema URL. Let me know if that is still needed.

**Comment by RichardBruskiewich** [comment]
Now available since commit 0c5059615d99fbdf6ba0ab8a68d6e9b288999b8d (release v3.3.1)
